SampleApp/ViewController.swift

```diff
@@ -7,62 +7,75 @@ import RxSwift
 import Starscream
 
 class ViewController: UIViewController {
-  
-  @IBOutlet weak var inputTextField: UITextField!
-  @IBOutlet weak var sendButton: UIButton!
-  @IBOutlet weak var logTextView: UITextView!
-  
-  let disposeBag = DisposeBag()
-  let socket = WebSocket(url: URL(string: "wss://echo.websocket.org")!)
-  
-  override func viewDidLoad() {
-    super.viewDidLoad()
+
+    @IBOutlet fileprivate weak var inputTextField: UITextField!
+    @IBOutlet fileprivate weak var sendButton: UIButton!
+    @IBOutlet fileprivate weak var logTextView: UITextView!
+    @IBOutlet fileprivate weak var pongButton: UIBarButtonItem!
+
+    private let disposeBag = DisposeBag()
+    private let socket = WebSocket(url: URL(string: "wss://echo.websocket.org")!)
+    private let writeSubject = PublishSubject<String>()
     
-    socket.rx.response
-      .subscribe(onNext: { [weak self] (response: WebSocketEvent) in
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        sendButton.rx.tap
+                .subscribe(onNext: { [unowned self] in
```

Comment:
Not sure if this in this case will be better.. It's just a UIButton and tap event.